eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/Utils.java

```diff
@@ -49,6 +49,11 @@ protected static boolean hasIllegalCombination(DetailAST modifiers) {
             return false;
         }
 
+        if (modifiers.getChildCount() == 0) {
+            // no modifier for field. Means it is using default and we don't want those to be final
```

Comment:
Why don't we want fields with a package-private modifier to be final?

Reply:
I think adding this logic creates a blindspot because it won't flag any more of these failures. And someone will have to come back and remove this code later, so who knows how many more package-private fields may have been added that aren't final.

I would create a GitHub issue for all the ones that fail, assign the issue to the owner, add a suppression line with the link to the issue and make sure the owner fixes/addresses the issue then removes/keeps the suppression.

Reply:
well, first we need to agree if this is actually an issue right? and what the issue actually is.

The final fields rule is: 'There is an issue if a field is not final and It is only assigned on class constructor or initialized and never reassigned again'

Following that rule, a package-private field won't break this rule! since it is meant to be accessed from other classes on same package to be reassigned right?  (some kind of public field within the package).

So, IMO, if we don't like this pattern, we need a new rule to avoid accessing fields directly and always use set/get methods for it. So, basically blocking public, protected and default fields to be used without static or final restriction.

let me know what you think @conniey @JonathanGiles

Reply:
I agree with Connie.

Reply:
issue created: https://github.com/Azure/azure-sdk-for-java/issues/5030

I will remove this validation